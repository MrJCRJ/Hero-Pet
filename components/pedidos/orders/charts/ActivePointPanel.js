import React from "react";
import { formatMoney, formatPercent } from "./shared/formatters";

/**
 * Painel reutilizÃ¡vel para exibir detalhes do ponto ativo em grÃ¡ficos de sÃ©ries temporais.
 * Props:
 *  - point: { label }
 *  - prevPoint: ponto anterior (ou null)
 *  - momPct: nÃºmero ou null (variaÃ§Ã£o percentual mÃªs a mÃªs)
 *  - acumuladaPct: nÃºmero (variaÃ§Ã£o acumulada desde o primeiro mÃªs)
 *  - rows: array de { label, value, type?: 'money' | 'percent' | 'raw' }
 *  - moneyFormatter?: fn custom (default formatMoney)
 *  - percentFormatter?: fn custom
 *  - icons?: { mom?: string, acumulado?: string }
 */
export default function ActivePointPanel({
  point,
  prevPoint,
  momPct,
  acumuladaPct,
  rows = [],
  moneyFormatter = formatMoney,
  percentFormatter = (n) => formatPercent(n, { withSign: true }),
  icons = { mom: "ðŸ“Š", acumulado: "ðŸ“ˆ" },
  showAcumulado = true,
  showPrev = true,
}) {
  if (!point) return null;

  const renderValue = (row) => {
    const { value, type } = row;
    if (type === "money")
      return <span className="font-semibold">{moneyFormatter(value)}</span>;
    if (type === "percent")
      return <span className="font-semibold">{percentFormatter(value)}</span>;
    return <span className="font-semibold">{value}</span>;
  };

  return (
    <div className="mt-3 p-3 bg-[var(--color-bg-secondary)] border rounded text-xs">
      <div className="font-medium mb-2">{point.label}</div>
      <div className="grid grid-cols-2 gap-3">
        {rows.map((row) => (
          <div key={row.label}>
            <div className="opacity-70">{row.label}</div>
            {renderValue(row)}
          </div>
        ))}
        {momPct != null && (
          <div>
            <div className="opacity-70">MoM {icons.mom}</div>
            <div
              className={`font-semibold ${momPct >= 0 ? "text-green-600" : "text-red-600"}`}
            >
              {percentFormatter(momPct)}
            </div>
          </div>
        )}
        {showPrev && prevPoint && (
          <div>
            <div className="opacity-70">MÃªs anterior</div>
            <div>{moneyFormatter(prevPoint.value)}</div>
          </div>
        )}
        {showAcumulado && (
          <div>
            <div className="opacity-70">Acumulado {icons.acumulado}</div>
            <div
              className={`${acumuladaPct >= 0 ? "text-green-600" : "text-red-600"}`}
            >
              {percentFormatter(acumuladaPct)}
            </div>
          </div>
        )}
      </div>
    </div>
  );
}
