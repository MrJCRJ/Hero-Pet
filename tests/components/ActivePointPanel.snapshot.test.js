import React from 'react';
import { render } from '@testing-library/react';
import ActivePointPanel from '../../components/orders/charts/ActivePointPanel';

const basePoint = { label: '2025-01', value: 1000 };
const prevPoint = { label: '2024-12', value: 800 };

describe('ActivePointPanel snapshots', () => {
  it('render completo', () => {
    const { container } = render(
      <ActivePointPanel
        point={basePoint}
        prevPoint={prevPoint}
        momPct={25}
        acumuladaPct={50}
        rows={[{ label: 'Vendas', value: 1000, type: 'money' }]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('sem acumulado', () => {
    const { container } = render(
      <ActivePointPanel
        point={basePoint}
        prevPoint={prevPoint}
        momPct={25}
        acumuladaPct={50}
        showAcumulado={false}
        rows={[{ label: 'Vendas', value: 1000, type: 'money' }]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('sem prev', () => {
    const { container } = render(
      <ActivePointPanel
        point={basePoint}
        prevPoint={null}
        momPct={25}
        acumuladaPct={50}
        showPrev={false}
        rows={[{ label: 'Vendas', value: 1000, type: 'money' }]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('sem mom', () => {
    const { container } = render(
      <ActivePointPanel
        point={basePoint}
        prevPoint={prevPoint}
        momPct={null}
        acumuladaPct={50}
        rows={[{ label: 'Vendas', value: 1000, type: 'money' }]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
