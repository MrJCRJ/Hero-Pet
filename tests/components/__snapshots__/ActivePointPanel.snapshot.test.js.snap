// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivePointPanel snapshots render completo 1`] = `
<div>
  <div
    class="mt-3 p-3 bg-[var(--color-bg-secondary)] border rounded text-xs"
  >
    <div
      class="font-medium mb-2"
    >
      2025-01
    </div>
    <div
      class="grid grid-cols-2 gap-3"
    >
      <div>
        <div
          class="opacity-70"
        >
          Vendas
        </div>
        <span
          class="font-semibold"
        >
          R$Â 1.000,00
        </span>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          MoM 
          ðŸ“Š
        </div>
        <div
          class="font-semibold text-green-600"
        >
          +25.0%
        </div>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          MÃªs anterior
        </div>
        <div>
          R$Â 800,00
        </div>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          Acumulado 
          ðŸ“ˆ
        </div>
        <div
          class="text-green-600"
        >
          +50.0%
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActivePointPanel snapshots sem acumulado 1`] = `
<div>
  <div
    class="mt-3 p-3 bg-[var(--color-bg-secondary)] border rounded text-xs"
  >
    <div
      class="font-medium mb-2"
    >
      2025-01
    </div>
    <div
      class="grid grid-cols-2 gap-3"
    >
      <div>
        <div
          class="opacity-70"
        >
          Vendas
        </div>
        <span
          class="font-semibold"
        >
          R$Â 1.000,00
        </span>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          MoM 
          ðŸ“Š
        </div>
        <div
          class="font-semibold text-green-600"
        >
          +25.0%
        </div>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          MÃªs anterior
        </div>
        <div>
          R$Â 800,00
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActivePointPanel snapshots sem mom 1`] = `
<div>
  <div
    class="mt-3 p-3 bg-[var(--color-bg-secondary)] border rounded text-xs"
  >
    <div
      class="font-medium mb-2"
    >
      2025-01
    </div>
    <div
      class="grid grid-cols-2 gap-3"
    >
      <div>
        <div
          class="opacity-70"
        >
          Vendas
        </div>
        <span
          class="font-semibold"
        >
          R$Â 1.000,00
        </span>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          MÃªs anterior
        </div>
        <div>
          R$Â 800,00
        </div>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          Acumulado 
          ðŸ“ˆ
        </div>
        <div
          class="text-green-600"
        >
          +50.0%
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActivePointPanel snapshots sem prev 1`] = `
<div>
  <div
    class="mt-3 p-3 bg-[var(--color-bg-secondary)] border rounded text-xs"
  >
    <div
      class="font-medium mb-2"
    >
      2025-01
    </div>
    <div
      class="grid grid-cols-2 gap-3"
    >
      <div>
        <div
          class="opacity-70"
        >
          Vendas
        </div>
        <span
          class="font-semibold"
        >
          R$Â 1.000,00
        </span>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          MoM 
          ðŸ“Š
        </div>
        <div
          class="font-semibold text-green-600"
        >
          +25.0%
        </div>
      </div>
      <div>
        <div
          class="opacity-70"
        >
          Acumulado 
          ðŸ“ˆ
        </div>
        <div
          class="text-green-600"
        >
          +50.0%
        </div>
      </div>
    </div>
  </div>
</div>
`;
